src/main/java/internetShop/dao/jdbc/ProductJdbcDao.java

```diff
@@ -2,13 +2,27 @@
 
 import internetShop.dao.ProductDao;
 import internetShop.entity.product.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.RowMapper;
-import org.springframework.jdbc.core.support.JdbcDaoSupport;
+import org.springframework.stereotype.Repository;
 
+import javax.sql.DataSource;
 import java.sql.PreparedStatement;
 import java.util.List;
 
-public class ProductJdbcDao extends JdbcDaoSupport implements ProductDao {
+@Repository
+public class ProductJdbcDao implements ProductDao {
+
+    private JdbcTemplate jdbcTemplate;
+    private DataSource dataSource;
+
+    @Autowired
+    public void setDataSource(@Qualifier("dataSource") DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
```

Comment:
you can define JdbcTemplate in JavaConfig set into dataSource

on a class use @RequiredArgs...

private final JdbcTemplate jdbcTemplate;